src/include/duckdb/planner/logical_operator.hpp

```diff
@@ -48,6 +48,9 @@ class LogicalOperator {
 	//! Estimated Cardinality
 	idx_t estimated_cardinality;
 	bool has_estimated_cardinality;
+	//! Flag to track if left and right child have been swapped in a join
+	//! Needed so that lateral_depth can be tracked and updated correctly during flattening and planning
+	bool swapped_children = false;
```

Comment:
I don't quite follow why we need this flag - it seems that this flag is only used for lateral joins, yet it is always false for lateral joins. Could we remove it or am I missing something?

Reply:
In the case of a right join, the optimizer swaps the left and right child to make it a left join. Without the `swapped_children` flag, `lateral_depth` would be incremented for the call to the left child (as it has now become the right child), which triggers assertions on correlated column depth (since there is no LateralBinder on the left child). As such, we set `lateral_depth = true` in plan_joinref.cpp, when a right join is detected and the optimizer is on.

Reply:
Yes, this flag is not needed.
It was an artifact of our previous design where we used to track all the joins (not just dependent joins). In that case, we had to specially handle the right joins as the bindings would be off when the two sides were swapped.
In the new design, we only track lateral depth and don't need this flag anymore.

Thanks for pointing that out, we missed it during our code cleanup.
Will push a commit to clean this up and the rest of the comments you have added.